### PR TITLE
Keep docstring consistent on supporting vector, array, and columnar input formats

### PR DESCRIPTION
**Pull request by lijinf2**
Removed featuresCols and inputCols in docstring as they have been included in featuresCol and inputCol.

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/clustering.py [hunk 1]
```diff
@@ -149,11 +149,10 @@ class KMeans(KMeansClass, _CumlEstimator, _KMeansCumlParams):
     tol: float (default = 1e-4)
         early stopping criterion if centers do not change much after an iteration.
 
-    featuresCol: str
-        the name of the column that contains input vectors. featuresCol should be set when input vectors are stored in a single column of a dataframe.
-
-    featuresCols: List[str]
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/clustering.py:155]: I see a Param with this name and a mixin (HasFeaturesCols) as well as getters and setters.   But it is not included as a constructor arg.   So are we getting rid of featuresCols throughout?   Or should we add as a constructor param.  @wbo4958 ?

Comment (leewyang) [reply python/src/spark_rapids_ml/clustering.py:155]: FWIW, I think this was originally trying to mirror the `HasInputCol` and `HasInputCols` [params](https://spark.apache.org/docs/latest/api/python/_modules/pyspark/ml/param/shared.html), but it is obviously redundant if `featuresCol` support both single str and list of str (which is a bit non-standard).  Anyhow, as long as we maintain compatibility w/ Pyspark code/usage, I think it's probably OK to remove.

Comment (wbo4958) [reply python/src/spark_rapids_ml/clustering.py:155]: Seems we're using featuresCols to store the feature names, seems we can't remove it.

**Comment by wbo4958** [comment]
LGTM.

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**